packages/lib/zuauth/src/zuauth.ts

```diff
@@ -135,8 +147,7 @@ export function constructZkTicketProofUrl(zuAuthArgs: ZuAuthArgs): string {
     },
     validEventIds: {
       argumentType: ArgumentTypeName.StringArray,
-      value:
-        eventIds.length !== 0 && eventIds.length <= 20 ? eventIds : undefined,
+      value: undefined,
```

Comment:
This is losing a feature which we advertized at Devconnect.  How sure are you that nobody is depending on it?
Does ZKTelegram use ZuAuth?  We definitely went our of our way to make the point that logging in to DMS or posting via ZuRat didn't reveal which Devconnect subevent you attended.  It would be odd to silently undo that.

It would be possible to keep using this field when you know it's sufficient (number of eventIDs is <=20, and no productIDs are used).  That being said, I don't like that it was an implicit feature before.  It would be better if it were an explicit configuration choice by the developer to use this feature, with the library throwing an exception if the dev tries to use this feature with an overlarge list.

Note that when we get to PODTickets we can have longer lists, and also triples of (key, eventID, productID), so maybe looking ahead to that is the right way to not worry too much about getting this perfect.

Reply:
ZuAuth was not used at Devconnect, because it didn't exist then. The underlying code in `ZkEdDSAEventTicketPCD` and `passport-interface` was used, and still permits the use of `validEventIds`.

My thinking was that `validEventIds` is not a great feature to expose - it's confusing because it only works for event IDs and not product IDs. The more features we expose, the more we need to test and to document, and the purpose of ZuAuth is specifically to reduce the API surface area when compared to using the underlying libraries directly.

That said, I can make it an optionally-configurable feature that is off by default, which seems like the best of both worlds.

Reply:
Hmm. After implementing this, I remember why I didn't want to do it: the `validEventIds` model is just fundamentally not helpful for authentication in a world with multiple signers. The signer is always revealed, but unless you can determine the connection between the signer and the event ID, you can't rule out the possibility of some event ID having been used by the "wrong" signer. Passing a `validEventIds` check is just never good enough for authentication purposes, which is why we can't rely on it.

Reply:
Understood.  The limitation of that feature to validEventIds does assume that the signing key and product ID are either always a single value (e.g. signing key for last year's events), or don't matter for authentication (e.g. the product IDs for devconnect).  It would be possible to check for those scenarios, or leave them up to the user to configure properly.  However given comments elsewhere I'm okay with removing this so long as it's not affecting the use cases which were built to depend on it (like ZuRat for Devconnect).  With PODTickets we can bring back a much more flexible version of this.